### ISSUE TITLE
Estimator Report in regression: plot doesn't display correctly

### ISSUE THREAD

**Issue by MarieS-WiMLDS** [probabl-ai/skore#1292]
### Describe the bug

The prediction_error plot doesn't display correctly in some cases.  

### Steps/Code to Reproduce

```python
from sklearn.datasets import make_regression
from sklearn.linear_model import LinearRegression
from skore import EstimatorReport, ComparisonReport

X, y = make_regression(n_targets=2)
estimator = LinearRegression()
report_reg = EstimatorReport(estimator, X_train=X, y_train=y, X_test=X, y_test=y)

comp_reg = ComparisonReport(reports=[report_reg, report_reg], n_jobs = -1)
comp_reg.metrics.plot.prediction_error()
```

### Expected Behavior

- we should see the whole axis
- as much as possible, the legend shouldn't overlap the axis

### Actual Behavior

![Image](https://github.com/user-attachments/assets/1278a3ab-8a0d-4bea-8dbd-34cd9f532e2a)

### Environment

```shell
System:
    python: 3.12.8 | packaged by Anaconda, Inc. | (main, Dec 11 2024, 16:31:09) [GCC 11.2.0]
executable: /home/marie/anaconda3/envs/skore_test/bin/python
   machine: Linux-6.8.0-52-generic-x86_64-with-glibc2.35

Python dependencies:
        skore: 0.0.0+unknown
          pip: 24.2
   setuptools: 75.1.0
    diskcache: 5.6.3
      fastapi: 0.115.6
       joblib: 1.4.2
   matplotlib: 3.9.3
        numpy: 2.2.0
       orjson: 3.10.15
       pandas: 2.2.3
 platformdirs: 4.3.6
       plotly: 5.24.1
       psutil: 5.9.0
      pyarrow: 18.1.0
         rich: 13.9.4
 scikit-learn: 1.6.0
        skops: 0.11.0
      uvicorn: 0.32.1

Current version of the branch comparator (as of Feb 6th morning), this piece of skore code corresponds to skore 0.6.1.
```

**Comment by auguste-probabl** [comment]
Notes while working on this:
- This is about `ComparisonReport`, which is in development in #1286 
- `make_regression(n_targets=2)` makes the example problem a multi-output regression, which we haven't really dealt with yet. However, the behaviour remains even when the parameter is removed.

**Comment by auguste-probabl** [comment]
What I've tried
- `ax.figure.set_size_inches(10, 10)`: Doesn't increase the height, just the width.
- `plt.subplots(layout="constrained")`: Doesn't change anything.

I'm going to leave it there for now as I'm not sure what else to try.

**Comment by glemaitre** [comment]
I have to check but it could be that we put a constrained "square" and that the error is that small that the y-axis is shrink.

One thing might be to check if we can relaxed this constraint (if it exist).
